src/providers/uploadHelpers/uploadObservation.js

```diff
@@ -86,8 +102,10 @@ const uploadObservation = async ( obsToUpload: Object, localObs: Object ) => {
         inatjs.observation_sounds
       );
     }
+    return "success";
   } catch ( e ) {
-    console.log( "couldn't upload observation: ", e );
+    console.log( "couldn't upload observation: ", JSON.stringify( e.response ) );
+    return "failure";
```

Comment:
IMO, this `catch` block is catching too much. The part of the above where we're not really certain about whether an exception will occur is just `await inatjs.observations.create`. Maybe the network's spotty, maybe the server is having problems, maybe the server responds with an error message, etc., so we should try/catch around that because the app can't really control for those conditions. But we *can* control pretty much everything else, so we should not try/catch around that stuff b/c that just obscures exceptions from bugs we can and should fix.